### DISCUSSION TITLE
move  database  from mssql to postgresql

### DISCUSSION THREAD

**Discussion by nico8948** [DependencyTrack/dependency-track#4396] [Q&A]
I'm trying to move the database from mssql to postgresql.   What 1've done:   export the mssql tables into csv files.  Created a new  postgresql database, change the connection string from mssql to postgresql.   Restart the dependeny track.   The postgres tables are filled  the vanilla  data, the application is working.    When trying to import de csv file (converted to sql file) starting with with PROJECT table with all related files doesn't show the projects.   My  questions,  is it possible?  and  anyone has experience?

**Comment by nscuro** [comment]
There's no pre-built migration script or anything. Before I became maintainer, around the v3.8 -> v4 migration, I used [pgloader](https://pgloader.io/) to migrate from MySQL to PostgreSQL.

I documented what I did here: https://gist.github.com/nscuro/1d32c6099a8a82f98f737fffa936b416#file-instructions-md

But by now it will be horribly out-of-date, of course. But maybe it helps anyway.

Note that we will provide migration scripts from the currently supported RDBMSes to PostgreSQL when Dependency-Track v5 reaches general availability. v4 will continue to support MSSQL.

**Comment by DrcKarim** [comment]
Yes, you can. Just follow these steps:

1 - Change the database configuration in the backend
Update the database configuration from H2 to PostgreSQL in the application.properties file. Modify the following settings to point to PostgreSQL, replacing the database name and user as needed:

```
# Required
alpine.worker.threads=0
alpine.worker.thread.multiplier=4
alpine.worker.pool.drain.timeout.duration=PT5S

# Required
alpine.data.directory=~/.dependency-track

# Required
alpine.watchdog.logging.interval=0

# Required
# Set to 'external' for PostgreSQL
alpine.database.mode=external

# Optional
alpine.database.port=5432

# Required
# PostgreSQL connection URL
alpine.database.url=jdbc:postgresql://localhost:5432/dtrack

# Required
# PostgreSQL driver class
alpine.database.driver=org.postgresql.Driver

# Optional
# Uncomment and provide path if using a custom JDBC driver
# alpine.database.driver.path=/path/to/postgresql-42.2.1.jar

# Required
# PostgreSQL username and password
alpine.database.username=dtrack
alpine.database.password=dtrack
```


2 - Run the project
Start the project, and all tables will be automatically generated in the PostgreSQL database. After that, you can import your own data.

I hope this helps!
